tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraShortLivedTokenRoutesContract.scala

```diff
@@ -260,4 +261,88 @@ trait LinagoraShortLivedTokenRoutesContract {
            |}""".stripMargin)
   }
 
+  @Test
+  def getTokenShouldFailWithRevokedLongLivedToken(): Unit = {
+    val longLivedTokenRequest: String =
+      s"""{
+         |  "using": ["urn:ietf:params:jmap:core", "com:linagora:params:long:lived:token"],
+         |  "methodCalls": [
+         |    ["LongLivedToken/set", {
+         |      "accountId": "$ACCOUNT_ID",
+         |      "create": {
+         |        "K38": {
+         |          "deviceId": "$DEVICE_ID"
+         |        }
+         |      }
+         |    }, "c1"]
+         |  ]
+         |}""".stripMargin
+
+    val longLivedTokenResponse: String = `given`()
+      .headers(getHeadersWith(BOB_BASIC_AUTH_HEADER))
+      .body(longLivedTokenRequest)
+    .when()
+      .post()
+    .`then`
+      .statusCode(HttpStatus.SC_OK)
+      .extract()
+      .body()
+      .asString()
+
+    val longLivedTokenId: String = (((Json.parse(longLivedTokenResponse) \\ "methodResponses" )
+      .head \\ "created")
+      .head \\ "id")
+      .head.asInstanceOf[JsString].value
+
+    val longLivedToken: String = (((Json.parse(longLivedTokenResponse) \\ "methodResponses" )
+      .head \\ "created")
+      .head \\ "token")
+      .head.asInstanceOf[JsString].value
+
+    Thread.sleep(10000)
+
+    val revokeRequest: String =
+      s"""{
+         |  "using": ["urn:ietf:params:jmap:core", "com:linagora:params:long:lived:token"],
+         |  "methodCalls": [
+         |    ["LongLivedToken/set", {
+         |      "accountId": "$ACCOUNT_ID",
+         |      "destroy": ["$longLivedTokenId"]
+         |    }, "c1"]
+         |  ]
+         |}""".stripMargin
+
+    `given`
+      .headers(getHeadersWith(BOB_BASIC_AUTH_HEADER))
+      .body(revokeRequest)
+    .when()
+      .post()
+    .`then`
+      .statusCode(HttpStatus.SC_OK)
+      .contentType(JSON)
+
+    Thread.sleep(10000)
+
+    val response: String = `given`()
+      .header(buildBearerTokenHeader(longLivedToken))
+      .queryParam("type", "shortLived")
+      .queryParam("deviceId", DEVICE_ID)
+    .when()
+      .basePath(BASE_PATH)
+      .get().prettyPeek()
+    .`then`()
+      .statusCode(SC_INTERNAL_SERVER_ERROR)
```

Comment:
Well somehow it still returns successfully a jwt token...

By debugging the code, I could see that the piece of code in the DAO inside the Mono that actually revokes the token never really gets executed. If I do a block on the revoke method int the DAO, it works... Some reactor annoying trick again, and I can't wrap my head around where it's wrong exactly...